plugins/lookup/hashi_vault.py

```diff
@@ -428,6 +440,22 @@ def auth_aws_iam_login(self):
             Display().warning("HVAC should be updated to version 0.9.3 or higher. Deprecated method 'auth_aws_iam' will be used.")
             self.client.auth_aws_iam(**params)
 
+    def auth_jwt(self, **kwargs):
+        role_id = kwargs.get('role_id', os.environ.get('VAULT_ROLE_ID', None))
+        if role_id is None:
+            raise AnsibleError("Authentication method jwt requires a role_id")
+
+        jwt = kwargs.get('jwt', os.environ.get('VAULT_JWT', None))
+        if jwt is None:
+            raise AnsibleError("Authentication method jwt requires parameter jwt to be set")
+
+        mount_point = kwargs.get('mount_point')
+        if mount_point is None:
+            mount_point = 'jwt'
```

Comment:
You'll need to add a `verify_` method, and these types of checks should go in there, see the other auth methods for examples.

Reply:
@briantist Do you mean a `validate_` method? I have looked through the recipe for adding new auth backends, so I am working on something. Is it acceptable that the `validate_` method has side-effects? It seems like some of the other `validate_`-methods are also setting parameter values.... I see at least two parameters that need special care:

1. The Vault Role parameter to the hvac `jwt_login` method is named `role`, and not `role_id`. But I think we should use the same parameter in this lookup plugin. So the `role_id` parameter will have to be "renamed" to `role`.

1. I think the Vault mount point should default to `jwt` when using JWT Auth. Is that taken care of, or do I need to code it?

Reply:
@briantist I have committed a new implementation - where I have tried to take your comments into account.